test/graphql/queries/categories_query_test.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# test/graphql/queries/wallets_query_test.rb
+
+require 'test_helper'
+
+module Queries
+  class CategoriesQueryTest < ActionDispatch::IntegrationTest
+    setup do
+      password = 'Password123!'
+      expected_email = 'info@kwartracker.com'
+      @user = create(:user,
+                     email: expected_email,
+                     password: password,
+                     password_confirmation: password)
+      post('/graphql',
+           params: {
+             query: sign_in_with_email_mutation,
+             variables: sign_in_with_email_mutation_variables({ email: @user.email,
+                                                                password: password })
+           })
+
+      @json_response = parse_graphql_response(response.body)
+      @token = @json_response.dig('signInWithEmail', 'token')
+    end
+
+    test 'fetch categories' do
+      post('/graphql',
+           params: {
+             query: categories_query
+           }, headers: {
+             'Authorization': "Bearer #{@token}"
+           })
+      @json_response = parse_graphql_response(response.body)
+      assert_equal @json_response['categories'].count, 2
```

Comment:
better to test the created categories(default after creation) so we know that the generation is done after creation of user